### ISSUE TITLE
Problem with CMAKE_BUILD_TYPE

### ISSUE THREAD

**Issue by MarkusFrankATcernch** [AIDASoft/DD4hep#839]
I'm currently facing cmake warnings when compiling LHCb software with
DD4hep support. You can find the exact output at
https://lhcb-nightlies.web.cern.ch/nightly/lhcb-dd4hep/836/LHCb/x86_64-centos7-gcc10-opt/build#configure_274

it's basically saying :

CMake Warning at /workspace/venv/centos7/lib/python2.7/site-packages/LbDevTools/data/cmake/GaudiBuildFlags.cmake:35 (message):
  CMAKE_BUILD_TYPE set to RelWithDebInfo, but BINARY_TAG build type dbg implies Debug

This seems to be due to the fact that you're forcing the CMAKE_BUILD_TYPE in DD4hepConfig.cmake.in :
https://github.com/AIDASoft/DD4hep/blob/4753d299daf1f81a48fc0dc84b26847abb31bd85/cmake/DD4hepConfig.cmake.in#L48
basically imposing it to downstream projects.

**Comment by petricm** [comment]
Yes, there is not really a good reason why we propagate this info. Thoughts @andresailer ?

**Comment by andresailer** [comment]
Yeah, we don't even propagate the info from how DD4hep was build, but actually always set RelWithDebInfo, so this doesn't make a lot of sense.
